Modules/Loadable/Colors/Logic/vtkSlicerColorLogic.cxx

```diff
@@ -209,3 +214,38 @@ std::vector<std::string> vtkSlicerColorLogic::FindColorFiles(const std::vector<s
     } // end of looping over dirs
   return filenames;
 }
+
+//----------------------------------------------------------------------------
+vtkMRMLColorBarDisplayNode* vtkSlicerColorLogic::CreateAndObserveColorBarForNode(vtkMRMLDisplayableNode* dispNode)
+{
+  if (!dispNode)
+  {
+    vtkDebugMacro("CreateAndObserveColorBarForNode: Displayable node is invalid");
+    return nullptr;
+  }
+
+  vtkMRMLColorBarDisplayNode* cbNode = vtkMRMLColorBarDisplayNode::SafeDownCast(
+    dispNode->GetNodeReference(vtkMRMLColorBarDisplayNode::COLOR_BAR_REFERENCE_ROLE));
+  if (cbNode)
+  {
+    vtkDebugMacro("CreateAndObserveColorBarForNode: Displayable node already has color bar node");
+    return cbNode;
+  }
+
+  vtkMRMLScene* mrmlScene = this->GetMRMLScene();
+  if (!mrmlScene)
+  {
+    vtkErrorMacro("CreateAndObserveColorBarForNode: Invalid MRML scene");
+    return nullptr;
+  }
+
+  // Create color bar and set reference
+  vtkNew<vtkMRMLColorBarDisplayNode> colorBarNode;
+  mrmlScene->AddNode(colorBarNode);
+  std::string uniqueName = mrmlScene->GenerateUniqueName("ColorBar");
+  colorBarNode->SetName(uniqueName.c_str());
+
+  dispNode->SetNodeReferenceID(vtkMRMLColorBarDisplayNode::COLOR_BAR_REFERENCE_ROLE, colorBarNode->GetID());
```

Comment:
The color bar is just another display node (similarly how the scalar volume node has additional display node(s) for volume rendering), so we don't need a special reference role. Instead, we should call:

    dispNode->AddAndObserveDisplayNodeID(colorBarNode->GetID());